src/main/java/hudson/plugins/sshslaves/SSHLauncher.java

```diff
@@ -862,12 +869,16 @@ public Boolean call() throws InterruptedException {
                 System.out.println(Messages.SSHLauncher_LaunchCompletedDuration(getTimestamp(),
                         nodeName, host, duration));
             }
-            executorService.shutdown();
```

Comment:
It was not shutting down the executor service when we were interrupting the launch :(